### PR TITLE
keycloak: 20.0.5 -> 21.1.1, keycloak-metrics-spi: 2.5.3 -> 3.0.0

### PR DESCRIPTION
**Pull request by ngerstle**
###### Description of changes

Based on https://github.com/NixOS/nixpkgs/pull/219010, with the added change to fix the metrics-spi plugin  to pass tests: https://github.com/NixOS/nixpkgs/pull/219010#issuecomment-1518587338

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by teutat3s — COMMENTED** [review]
Thank you for this. Could you rebase to version 21.1.1 and add the changelogs to the commit, while you're at it? Nit: maybe split the two updates into two separate commits.

**Comment by ngerstle** [comment]
bundles https://github.com/NixOS/nixpkgs/pull/219010 as well

**Review by teutat3s — CHANGES_REQUESTED** [review]

File: pkgs/servers/keycloak/keycloak-metrics-spi/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 stdenv.mkDerivation rec {
   pname = "keycloak-metrics-spi";
-  version = "2.5.3";
+  version = "2.3.0";
```
[the remark this answers is not in the document]
Comment (teutat3s) [reply pkgs/servers/keycloak/keycloak-metrics-spi/default.nix:5]: While you fix this, please add the [changelog](https://github.com/aerogear/keycloak-metrics-spi/releases/tag/3.0.0) to the commit message as well.

[the remark this answers is not in the document]
Comment (teutat3s) [reply pkgs/servers/keycloak/keycloak-metrics-spi/default.nix:5]: With this change, `nix-build nixos/tests/keycloak.nix` looks green here.

**Review by teutat3s — APPROVED** [review]

**Comment by teutat3s** [comment]
Result of `nixpkgs-review pr 229344` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 test built:</summary>
  <ul>
    <li>nixosTests.keycloak.postgres</li>
  </ul>
</details>

**Comment by talyz** [comment]
Tests pass, can't find any issues when deploying to my test environment. Thanks!

**Pull request merged**